build.gradle

```diff
@@ -66,16 +65,22 @@ dependencies {
     testImplementation 'org.jenkins-ci.plugins.workflow:workflow-job:1180.v04c4e75dce43'
     testImplementation 'org.jenkins-ci.plugins.workflow:workflow-step-api:625.vd896b_f445a_f8:tests'
     testImplementation "org.jenkins-ci.main:jenkins-war:${jenkinsVersion}"
-    testImplementation('org.testcontainers:spock:1.17.3') {
+    testImplementation('org.testcontainers:spock:1.17.4') {
         exclude group: 'org.spockframework', module: 'spock-core'
     }
-    testImplementation 'org.spockframework:spock-core:2.1-groovy-3.0'
+    testImplementation 'org.spockframework:spock-core:2.3-groovy-3.0'
+    testImplementation 'org.spockframework:spock-junit4:2.3-groovy-3.0'
     testImplementation 'junit:junit:4.13.2'
```

Comment:
There are two junit testimplementations? What about cross-effects?

Reply:
spock-core 2.x does not support junit 4 natively anymore. Therefore they created the spock-junit4 package which provides a compat layer.